lib/bundle-validator.js

```diff
@@ -402,6 +404,42 @@ class BundleValidator {
         return true;
     }
 
+    async _validateCssFiles() {
+        const assetsPath = path.join(this.themePath, 'assets');
+        const stylesPath = path.join(this.themePath, 'assets/scss');
+        const rawConfig = await this.themeConfig.getRawConfig();
+        const compiler = rawConfig.css_compiler;
+
+        let cssFiles;
+        try {
+            const files = await fs.promises.readdir(stylesPath);
+            cssFiles = files.filter((file) => {
+                return file.slice(-(compiler.length + 1)) === `.${compiler}`;
+            });
+        } catch (e) {
+            throw new Error(`Error: ${e.message}, while reading files from "${stylesPath}".`.red);
+        }
+
+        for await (const file of cssFiles) {
```

Comment:
Does the use of `for await` here mean that `cssFiles` is an  async interable object? It seems like we're doing `await` inside of the `try` so dose that mean the filter/slice operation results in `async`?

Reply:
I think it was a leftover from previous solution. But you are right, that `for await` should be used for async iterable objects. On other hand it also supports `sync` iterables, I think it's cleaner to separate those concerns

Reply:
I think the problem was in the eslint as well: https://eslint.org/docs/rules/no-await-in-loop
But those operation steps should go in sync way, so I disabled that rule for the next line